internal/services/postgresqlhsc/postgresql_hyperscale_cluster_resource.go

```diff
@@ -0,0 +1,628 @@
+	PreferredPrimaryZone             string              `tfschema:"preferred_primary_zone"`
+	SqlVersion                       string              `tfschema:"sql_version"`
+	Tags                             map[string]string   `tfschema:"tags"`
+	EarliestRestoreTime              string              `tfschema:"earliest_restore_time"`
+}
+
+type MaintenanceWindow struct {
+	DayOfWeek   int64 `tfschema:"day_of_week"`
+	StartHour   int64 `tfschema:"start_hour"`
+	StartMinute int64 `tfschema:"start_minute"`
+}
+
+type PostgreSQLHyperScaleClusterResource struct{}
+
+var _ sdk.ResourceWithUpdate = PostgreSQLHyperScaleClusterResource{}
+
+func (r PostgreSQLHyperScaleClusterResource) ResourceType() string {
+	return "azurerm_postgresql_hyperscale_cluster"
+}
+
+func (r PostgreSQLHyperScaleClusterResource) ModelObject() interface{} {
+	return &PostgreSQLHyperScaleClusterModel{}
+}
+
+func (r PostgreSQLHyperScaleClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return clusters.ValidateServerGroupsv2ID
+}
+
+func (r PostgreSQLHyperScaleClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ClusterName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"administrator_login_password": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ValidateFunc: validate.AdministratorLoginPassword,
+		},
+
+		"coordinator_storage_quota_in_mb": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(32768, 16777216),
+				validation.IntDivisibleBy(1024),
+			),
+		},
+
+		"coordinator_vcores": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ValidateFunc: validation.IntInSlice([]int{
+				1,
+				2,
+				4,
+				8,
+				16,
+				32,
+				64,
+				96,
+			}),
+		},
+
+		"node_count": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(0, 20),
+				validation.IntNotInSlice([]int{1}),
+			),
+		},
+
+		"citus_version": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"8.3",
+				"9.0",
+				"9.1",
+				"9.2",
+				"9.3",
+				"9.4",
+				"9.5",
+				"10.0",
+				"10.1",
+				"10.2",
+				"11.0",
+				"11.1",
+				"11.2",
+			}, false),
+		},
+
+		"coordinator_public_ip_access_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"coordinator_server_edition": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Default:  "GeneralPurpose",
+			ValidateFunc: validation.StringInSlice([]string{
+				"BurstableGeneralPurpose",
+				"BurstableMemoryOptimized",
+				"GeneralPurpose",
+				"MemoryOptimized",
```

Comment:
are there SDK constants for these values?

Reply:
I think there there are no SDK constants for these values. These values are provided by service team.

Reply:
@tombuildsstuff , opened a [swagger issue](https://github.com/Azure/azure-rest-api-specs/issues/23317) for it. Once the issue is fixed, I will submit another PR to improve the validation. So could we continue to review and merge this PR in parallel? Thanks.

Reply:
@neil-yechenwei [this comment](https://github.com/hashicorp/terraform-provider-azurerm/issues/5040#issuecomment-1487274763) sounds like this service may be end-of-life, can you confirm with the service team?

Reply:
@tombuildsstuff , service team confirmed they have transited to "CosmosDB PostgreSQL" and the [API](https://github.com/Azure/azure-rest-api-specs/blob/0f7cb051d05db783cfbaa50c864594a9d50e6f91/specification/postgresqlhsc/resource-manager/Microsoft.DBforPostgreSQL/stable/2022-11-08/postgresqlhsc.json#L4) we are using in this PR is used for "CosmosDB PostgreSQL". So I updated the name accordingly. Please take another look. Thanks.